packages/celer-web-app/src/ui/frames/DocFrame.tsx

```diff
@@ -44,15 +45,75 @@ const centerMapToLine = (docLine: DocLineText | DocLineTextWithIcon, setMapCente
 	}
 };
 
+const scrollToLine = (targetLine: number, docLineRefs: React.RefObject<HTMLDivElement>[], setCurrentLine: React.Dispatch<React.SetStateAction<number>>) => {
+	docLineRefs[targetLine].current?.scrollIntoView();
+	setCurrentLine(targetLine);
+};
+
+const findSplit = (currentLine: number, docLines: DocLine[], docLineRefs: React.RefObject<HTMLDivElement>[], splitSettings: SplitTypeSetting<boolean>, setCurrentLine: React.Dispatch<React.SetStateAction<number>>, up: boolean) => {
+	let temp = currentLine;
+	if(up){
+		if(temp!=0){
+			temp--;
+		}
+	}else{
+		if(temp<docLines.length){
+			temp++;
+		}
+	}
+	while(temp>0 && temp<docLines.length-1){
+		const checkedLine= docLines[temp];
+		if(checkedLine.lineType==="DocLineTextWithIcon"){
+			if(splitSettings[checkedLine.splitType]){
+				break;
+			}
+		}
+		if(up){
+			temp--;
+		}else{
+			temp++;
+		}
+	}
+	scrollToLine(temp, docLineRefs, setCurrentLine);
+};
+
 export const DocFrame: React.FC<DocFrameProps> = ({docLines})=>{
 	const [updateHandle, setUpdateHandle] = useState<number|undefined>(undefined);
 	const {setDocCurrentLine, setMapCenter} = useAppState();
 	const ScrollProgressTrackerEnabled = useExpScrollProgressTrackerEnabled();
+	const [currentLine, setCurrentLine] = useState(0);
+	const {setting} = useAppSetting();
 	const docFrameRef = useRef<HTMLDivElement>(null);
-
 	// Loading styles
 	const styles = useStyles();
 
+	const keyPressed = (e: KeyboardEvent) => {
```

Comment:
You can move this function inside the useEffect hook below. Currently it's not obvious why the useEffect below depends on setting and currentLine

Reply:
Javascript seems to predefine the function on the keyboard listener, which then does not update with currentline and settings for split settings. The effect is dependent on those values to update the keyboard listener. I did move the definition for the function into the effect though.